docs/index.html

```diff
@@ -20,396 +20,490 @@
             height: 2rem;
             animation: spin 1s infinite linear;
         }
+        .floating-spinner{
+            position: fixed;
+            top: 0;
+            left: 0;
+            width: 100%;
+            height: 100%;
+            display: flex;
+            justify-content: center;
+            align-items: center;
+            background-color: rgba(0, 0, 0, 0.1); /* Optional: for a semi-transparent background */
+            z-index: 9999 /* Ensure it is on top of other elements */
+
+        }
+
         @keyframes spin {
             to {
                 transform: rotate(360deg);
             }
         }
+
         .tab-active {
             background-color: var(--color-turquoise);
             border-radius: 0.5rem;
             color: white;
         }
+
         .separator {
             border-top: 1px solid #555;
             margin: 1rem 0;
         }
     </style>
     <script src="./worker-manager.js"></script>
 </head>
+
 <body>
-<div id="app"></div>
-
-<script type="text/babel" data-presets="env,react">
-    const { useState, useEffect } = React;
-
-    const PYTHON_FILE_NAME = 'provide_images.py';
-
-    function App() {
-        const [serverURL, setServerURL] = useState("wss://hypha.aicell.io/ws");
-        const [pyodideWorker, setPyodideWorker] = useState(null);
-        const [code, setCode] = useState("");
-        const [user, setUser] = useState({});
-        const [imageFolderHandle, setImageFolderHandle] = useState(null);
-        const [imageList, setImageList] = useState([]);
-        const [annotationsFolderHandle, setAnnotationsFolderHandle] = useState(null);
-        const [annotationsList, setAnnotationsList] = useState([]);
-        const [isLoadingWorker, setIsLoadingWorker] = useState(true);
-        const [isLoadingImages, setIsLoadingImages] = useState(false);
-        const [isLoadingAnnotations, setIsLoadingAnnotations] = useState(false);
-        const [annotationURL, setAnnotationURL] = useState("");
-        const [copyFeedback, setCopyFeedback] = useState("");
-        const [activeTab, setActiveTab] = useState("Local Deployment");
-
-        useEffect(() => {
-            const workerManager = new PyodideWorkerManager();
-            workerManager.createWorker().then(worker => {
-                setPyodideWorker(worker);
-                setIsLoadingWorker(false); // Hide spinner when worker is ready
-            }).catch(error => {
-                console.error("Error creating Pyodide worker:", error);
-            });
-
-            fetch(`./${PYTHON_FILE_NAME}`)
-                .then(response => response.text())
-                .then(text => {
-                    setCode(text);
-                })
-                .catch(error => {
-                    console.error(`Error loading ${PYTHON_FILE_NAME}:`, error);
-                });
-        }, []);
+    <script type="module">
+        import { get, set } from 'https://cdn.jsdelivr.net/npm/idb-keyval@6/+esm';
```

Comment:
@nilsmechtel This is for saving file handle into indexeddb, so we don't need to select the folder every time when the page reload